src/middlewared/middlewared/plugins/jail.py

```diff
@@ -214,6 +214,51 @@ def create_job(self, job, options):
 
         return True
 
+    @accepts(
+        Str('source_jail', empty=False),
+        Dict(
+            'clone_jail',
+            Str('uuid', required=True, empty=False),
+            List('pkglist', default=[], items=[Str('pkg', empty=False)]),
+            Bool('thickjail', default=False),
+            List('props', default=[]),
+        )
+    )
+    @job(lock=lambda args: f'clone_jail:{args[-2]}')
+    def clone_jail(self, job, source_jail, options):
```

Comment:
Just `clone` is OK, we are already in jail namespace.

We probably want `@item_method` decorator here.